### PR TITLE
bpo-40937: Remove deprecated universal mode

### PR DESCRIPTION
**Pull request by remilapeyre**
https://bugs.python.org/issue40937

### CODE REVIEW

File: Lib/_pyio.py [hunk 1]
```diff
@@ -87,10 +87,6 @@ def open(file, mode="r", buffering=-1, encoding=None, errors=None,
     returned as strings, the bytes having been first decoded using a
     platform-dependent encoding or using the specified encoding if given.
 
-    'U' mode is deprecated and will raise an exception in future versions
```
Comment (lrjball) [inline Lib/_pyio.py:90]: It wont let me add a comment to the line as it is too far away from the changes, but shouldn't line 74 also be removed here?

**Comment by lrjball** [comment]
Also, should the following line in gzip also be updated?

```python
if mode and ('t' in mode or 'U' in mode):
```
https://github.com/remilapeyre/cpython/blob/838a92a84813641c8a2ef936264d4ba30244d600/Lib/gzip.py#L168

**Comment by remilapeyre** [comment]
> Also, should the following line in gzip also be updated?
> 
> ```python
> if mode and ('t' in mode or 'U' in mode):
> ```
> 
> https://github.com/remilapeyre/cpython/blob/838a92a84813641c8a2ef936264d4ba30244d600/Lib/gzip.py#L168

Yes thanks!

**Comment by iritkatriel** [comment]
This has been removed in https://github.com/python/cpython/issues/81511

**Pull request closed without merging**